tests/catch/unit/memory/CMakeLists.txt

```diff
@@ -183,5 +183,4 @@ endif()
 
 hip_add_exe_to_target(NAME MemoryTest
                       TEST_SRC ${TEST_SRC}
-                      TEST_TARGET_NAME build_tests
-                      COMPILE_OPTIONS -std=c++14)
+                      TEST_TARGET_NAME build_tests)
```

Comment:
Any reason why we need to remove c++14 here?

Reply:
I noticed that tests/catch/CMakeLists sets the C++ version as C++17, with the stated intent in a comment being "enforce C++17 for all tests". The removed line overrode that setting for memory tests. 
In addition, this pull request uses a fold expression in DeviceAttributesSupport, and a constexpr lambda, both C++17 features. Future pull requests will use features like if constexpr and structured bindings as well.

Reply:
Thanks for the clarification. I am resolving some of the build failures seen by removing this flag override. Once it is done, this PR should pass tests and be ready to merge.